frb_example/with_flutter/lib/ffi.web.dart

```diff
@@ -0,0 +1,21 @@
+// ignore_for_file: avoid_web_libraries_in_flutter
+
+import 'dart:html';
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+import 'bridge_generated.dart';
+
+@JS()
+external WasmModule get wasmBindgen;
+
+Future<WasmModule> _initModule() {
+  if (crossOriginIsolated != true) return Future.error(MissingHeaderException());
+
+  final script = ScriptElement()..src = 'pkg/flutter_rust_bridge_example.js';
+  document.head!.append(script);
+  return script.onLoad.first.then((_) {
+    eval("window.wasmBindgen = wasm_bindgen");
+    return ([String? module]) => wasmBindgen(module ?? 'pkg/flutter_rust_bridge_example_bg.wasm');
+  });
+}
+
+final api = FlutterRustBridgeExampleImpl(_initModule() as ExternalLibrary);
```

Comment:
the `as` looks a bit weird: `Future<WasmModule>` is converted to an `ExternalLibrary`?

Reply:
Yeah, this is because ExternalLibrary appears in the constructor of the implementation class (no extensions for constructors) but it also has to be platform-dependent so it is defined as such in the support library. There should be a better way to express this.

Reply:
I see, it is completely reasonable.

Just wondering, if you change `typedef ExternalLibrary = FutureOr<WasmModule>;` to `typedef ExternalLibrary = Future<WasmModule>;`, will you be able to remove that `as` check?

Removing the check makes the code a bit nicer I guess

Reply:
A bit late, but not really since the user will always see `ExternalLibrary` as `DynamicLibrary`. It's only when compiling that it makes a difference.